OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -126,6 +126,45 @@ List<IFinalizedRenderable> GenerateRenderables()
 			return renderables;
 		}
 
+		List<IFinalizedRenderable> GenerateOverlayRenderables(IEnumerable<Actor> actorsInBox)
+		{
+			var aboveShroud = World.ActorsWithTrait<IRenderAboveShroud>().Where(a => a.Actor.IsInWorld && !a.Actor.Disposed)
```

Comment:
Introducing a new interface type for these still feels really ugly.

At the cost of a bit of perf savings, IMO it would be cleaner to instead:
* Add a `bool Partitionable { get }` property to `IRenderAboveShroud`
* Pass in `actorsInBox` as a `HashSet<Actor>`
* Change the `Where` query here to `a => a.Actor.IsInWorld && !a.Actor.Disposed && (!a.Trait.Partionable || actorsInBox.Contains(a))`

@RoosterDragon: what do you think?

Reply:
Sounds reasonable to me. I'll let @reaperrr decide on where to take things.